### ISSUE TITLE
Access to path error in `mkDummySrc`

### ISSUE THREAD

**Issue by yusdacra** [ipetkov/crane#10]
Hello, thanks for this great library! I'm implementing `crane` as a builder for https://github.com/nix-community/dream2nix, and I ran into an issue. Essentially I call `buildDepsOnly` and then use it in `cargoBuild`, but this doesn't seem to work because Nix complains about `error: access to absolute path '/Cargo.toml' is forbidden in pure eval mode (use '--impure' to override)`. I'm not sure why it's trying to access there, but this happens in the dummy source creation.

https://github.com/nix-community/dream2nix/blob/1c4b8aaa52e1e9572f590f2fdb099aeaf1734084/src/builders/rust/crane/default.nix#L32-L48 is where i use `builldDepsOnly`.

You can reproduce this with this flake:
```nix
{
  inputs.dream2nix.url = "github:yusdacra/dream2nix/rust/crane-builder";
  inputs.ripgrep = { flake = false; url = "github:BurntSushi/ripgrep/13.0.0"; };
  outputs = { self, nixpkgs, dream2nix, ripgrep }@inputs:
    let
      # change according to your system
      system = "x86_64-linux";
      dream2nix = inputs.dream2nix.lib.init {
        pkgs = nixpkgs.legacyPackages."${system}";
      };
    in
    {
      packages = (dream2nix.riseAndShine {
        source = ripgrep;
      }).packages;
    };
}
```

**Comment by ipetkov** [comment]
Hi @yusdacra, thanks for the report and the reproduction!

Seems like Nix might be evaluating path fragments way too eagerly (e.g. `src + /Cargo.toml` vs `src + "/Cargo.toml"`). Not sure why this is coming up, but with #11 it seems to make the problem go away (well, I ran into a _different_ problem, but it may be due to the dream2nix integration, which I'm not familiar enough with to debug further right now!)

**Comment by ipetkov** [comment]
Merged #10, I'm going to resolve this for now since I think _this_ immediate problem is resolved. Feel free to reopen if its not (or file a new issue if you find something else)!

**Comment by yusdacra** [comment]
> Seems like Nix might be evaluating path fragments way too eagerly (e.g. src + /Cargo.toml vs src + "/Cargo.toml"). Not sure why this is coming up, but with https://github.com/ipetkov/crane/pull/11 it seems to make the problem go away (well, I ran into a different problem, but it may be due to the dream2nix integration, which I'm not familiar enough with to debug further right now!)

I see, thanks for the quick fix! I updated my `crane` source and fixed the other problem, it seems to work well now :tada:
